src/spyglass/spikesorting/analysis/v1/unit_annotation.py

```diff
@@ -0,0 +1,150 @@
+        unit_key = {
+            "spikesorting_merge_id": key["spikesorting_merge_id"],
+            "unit_id": key["unit_id"],
+        }
+        if not self & unit_key:
+            nwb_file = (
+                SpikeSortingOutput & {"merge_id": key["spikesorting_merge_id"]}
+            ).fetch_nwb()[0]
+            nwb_field_name = self._get_spike_obj_name(nwb_file)
+            spikes = nwb_file[nwb_field_name]["spike_times"].to_list()
+            if key["unit_id"] > len(spikes):
+                raise ValueError(
+                    f"unit_id {key['unit_id']} is greater than ",
+                    f"the number of units in {key['spikesorting_merge_id']}",
+                )
+            self.insert1(unit_key)
+        # add annotation
+        self.Annotation().insert1(key, **kwargs)
+
+    def fetch_unit_spikes(self, return_unit_ids=False):
+        """Fetch the spike times for a restricted set of units
+
+        Parameters
+        ----------
+        return_unit_ids : bool, optional
+           whether to return unit ids with spike times, by default False
+
+        Returns
+        -------
+        list of np.ndarray
+            list of spike times for each unit in the group,
+            if return_unit_ids is False
+        tuple of list of np.ndarray, list of str
+            list of spike times for each unit in the group and the unit ids,
+            if return_unit_ids is True
+        """
+        if len(self) == len(UnitAnnotation()):
+            logger.warning(
+                "fetching all unit spikes",
+                "if this is unintended, please call as: ",
+                "(UnitAnnotation & key).fetch_unit_spikes()",
+            )
+        # get the set of nwb files to load
+        merge_keys = [
+            {"merge_id": merge_id}
+            for merge_id in list(set(self.fetch("spikesorting_merge_id")))
+        ]
+        nwb_file_list, merge_ids = (SpikeSortingOutput & merge_keys).fetch_nwb(
+            return_merge_ids=True
+        )
+
+        spikes = []
+        unit_ids = []
+        for nwb_file, merge_id in zip(nwb_file_list, merge_ids):
+            nwb_field_name = self._get_spike_obj_name(nwb_file)
+            sorting_spike_times = nwb_file[nwb_field_name][
+                "spike_times"
+            ].to_list()
+            include_unit = np.unique(
+                (self & {"spikesorting_merge_id": merge_id}).fetch("unit_id")
+            )
+            spikes.extend(
+                [sorting_spike_times[unit_id] for unit_id in include_unit]
+            )
+            unit_ids.extend(
+                [f"{merge_id}_{unit_id}" for unit_id in include_unit]
+            )
+
+        if return_unit_ids:
+            return spikes, unit_ids
+        return spikes
+
+    @staticmethod
+    def _get_spike_obj_name(nwb_file):
+        nwb_field_name = (
+            "object_id"
+            if "object_id" in nwb_file
+            else "units" if "units" in nwb_file else None
```

Comment:
This logic looks familiar - maybe in the spikesorting merge table? I wonder if there's a good way to centralize the terminology decision to avoid the knowledge of what it's called being this far downstream